File: DevAtlas Map/backend/app/routers/projects.py
import uuid

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.deps import get_db
from app.schemas.project import ProjectCreate, ProjectUpdate, ProjectResponse
from app.services import project_service

router = APIRouter(prefix="/projects", tags=["projects"])


@router.post("/", response_model=ProjectResponse, status_code=201)
async def create_project(data: ProjectCreate, db: AsyncSession = Depends(get_db)):
    return await project_service.create_project(db, data)


@router.get("/", response_model=list[ProjectResponse])
async def list_projects(db: AsyncSession = Depends(get_db)):
    return await project_service.get_projects(db)


@router.get("/{project_id}", response_model=ProjectResponse)
async def get_project(project_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    return await project_service.get_project(db, project_id)


@router.patch("/{project_id}", response_model=ProjectResponse)
async def update_project(project_id: uuid.UUID, data: ProjectUpdate, db: AsyncSession = Depends(get_db)):
    return await project_service.update_project(db, project_id, data)


@router.delete("/{project_id}", status_code=204)
async def delete_project(project_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    await project_service.delete_project(db, project_id)

File: AKW/frontend/src/api/projects.ts
import type { Project } from '@/types'
import client from './client'

export const projectsApi = {
  list: () => client.get<Project[]>('/projects/').then((r) => r.data),
  get: (id: string) => client.get<Project>(`/projects/${id}`).then((r) => r.data),
  create: (data: { name: string; description?: string; creator: string }) =>
    client.post<Project>('/projects/', data).then((r) => r.data),
  update: (id: string, data: { name?: string; description?: string }) =>
    client.patch<Project>(`/projects/${id}`, data).then((r) => r.data),
  delete: (id: string) => client.delete(`/projects/${id}`),
}
